src/hacbs/utils/commits-utils.ts

```diff
@@ -0,0 +1,62 @@
+import { PipelineRunKind, Commit } from '../types';
+
+export const COMMIT_ANNOTATION = 'build.appstudio.redhat.com/commit_sha';
+export const COMMIT_URL_ANNOTATION = 'build.appstudio.redhat.com/sha_url';
+export const COMMIT_BRANCH_ANNOTATION = 'build.appstudio.redhat.com/target_branch';
+export const COMMIT_COMPONENT_LABEL = 'build.appstudio.redhat.com/component';
+export const COMMIT_USER_LABEL = 'build.appstudio.redhat.com/user';
+export const COMMIT_REPO_URL = 'build.appstudio.redhat.com/sha_url';
+
+export const getCommitsFromPLRs = (plrList: PipelineRunKind[], limit?: number): Commit[] => {
+  const commits: Commit[] = [];
+  plrList.map((plr) => {
+    const commitSHA = plr.metadata.annotations?.[COMMIT_ANNOTATION];
+    const commitBranch = plr.metadata.annotations?.[COMMIT_BRANCH_ANNOTATION];
+    const commitUser = plr.metadata.labels?.[COMMIT_USER_LABEL];
+    const shaURL = plr.metadata.labels?.[COMMIT_URL_ANNOTATION];
+    const creationTime = plr.metadata.creationTimestamp;
+    const component = plr.metadata.labels[COMMIT_COMPONENT_LABEL];
+    const repoURL = plr.metadata.annotations[COMMIT_REPO_URL];
+
+    if (commitSHA) {
+      const existingCommitIndex = commits.findIndex((commit) => commit.sha === commitSHA);
+      if (existingCommitIndex > -1) {
+        if (commits[existingCommitIndex].creationTime < creationTime) {
+          commits[existingCommitIndex].creationTime = creationTime;
+        }
+        if (!commits[existingCommitIndex].component) {
+          commits[existingCommitIndex].component = component;
+        }
```

Comment:
As per design [doc](https://docs.google.com/document/d/1vlTQOmdwxHe6zMMLHjMjQ0Mi66uoR7_gPECbNP4d8T4/edit#heading=h.xes6l26uie9e), Components is a comma separated string if multiple components are involved.